pylean/pylean/__init__.py

```diff
@@ -0,0 +1,37 @@
+import pexpect
+import json
+
+import os
+
+class LeanServer:
+    def __init__(self):
+        path_to_repl = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+
+        self.proc = pexpect.spawn(
+            "lake env lean --run REPL/Main.lean", cwd=path_to_repl, encoding="utf-8"
+        )
+
+    def run_code(self, code, env=None, verbose=False):
+        if env:
+            command = (
+                '{ "cmd" : "' + repr(code)[1:-1] + f'", "env" : {env}' + " }"
+            )  # [1:-1] removes single quotes
+        else:
+            command = (
+                '{ "cmd" : "' + repr(code)[1:-1] + '" }'
+            )  # [1:-1] removes single quotes
+
+        if verbose:
+            print(command)
+        self.proc.sendline(command)
+        self.proc.expect_exact(command + "\r\n")
+        self.proc.sendline()
+        self.proc.expect_exact("\r\n")
```

Comment:
Does this work on unix? I wouldn't expect there to be an `\r` there

Reply:
This works on my ubuntu system.

Reply:
I believe the \r\n is part of JSON-RPC which is the basis for LSP